Sources/Jetworking/Client/Client.swift

```diff
@@ -143,16 +148,18 @@ public final class Client {
         urlResponse: URLResponse?,
         error: Error?,
         endpoint: Endpoint<ResponseType>,
-        completion: @escaping (Result<ResponseType, Error>) -> Void
+        completion: @escaping RequestCompletion<ResponseType>
     ) {
         var currentURLResponse = urlResponse
         configuration.responseInterceptors.forEach { component in
             currentURLResponse = component.intercept(data: data, response: currentURLResponse, error: error)
         }
 
-        if let error = error { return completion(.failure(error)) }
+        if let error = error { return completion(nil, .failure(error)) }
```

Comment:
Maybe we should move this behind the `guard` check if the urlResponse was not nil so we can pass the header values as well.

Reply:
Sure!